### PR TITLE
[FLINK-23009][kinesis] Upgrade Guava for Flink Connector Kinesis

### PR DESCRIPTION
**Pull request by iemre**
## What is the purpose of the change

Upgrade Guava dependency for Flink Connector Kinesis. The connector was using Guava 18.0 which is quite old and has been flagged by some security tools as noted in another ticket https://issues.apache.org/jira/browse/FLINK-22774 

This PR also makes the flink-sql-connector-kinesis use the Guava dependency transitively through the flink-connector-kinesis dependency.

## Brief change log

- Bumped Guava for Flink Connector Kinesis from 18.0 (inherited from parent) to 29.0-jre

## Verifying this change

This change is already covered by existing tests, such as `FlinkKinesisProducerTest`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes 
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by dannycranmer** [comment]
@iemre please update the NOTICE file for the SQL client

File: flink-connectors/flink-sql-connector-kinesis/pom.xml [hunk 1]
```diff
@@ -35,17 +35,6 @@ under the License.
...
-	<dependencyManagement>
-		<dependencies>
-			<dependency>
-				<!-- Bumped for security purposes to a common version in the Hadoop ecosystem -->
-				<groupId>com.google.guava</groupId>
-				<artifactId>guava</artifactId>
-				<version>27.0-jre</version>
-			</dependency>
-		</dependencies>
-	</dependencyManagement>
```
Comment (iemre) [inline flink-connectors/flink-sql-connector-kinesis/pom.xml:47]: the `flink-connector-kinesis` dependency (which is a part of the flink-connectors project) further below already declares Guava in its `pom.xml`. So we can get Guava transitively through `flink-connector-kinesis` and manage the Guava dependency in `flink-connector-kinesis/pom.xml` in a "one source of truth" way

Comment (dannycranmer) [reply flink-connectors/flink-sql-connector-kinesis/pom.xml:47]: Thanks 👍

**Review by dannycranmer — APPROVED** [review]
LGTM

**Pull request merged**